src/components/keyboard/virtual-keyboard.ts

```diff
@@ -0,0 +1,76 @@
+import type { Cell } from "@/components/GameCell.vue";
+import { reactive, watch } from "vue";
+import { useGrid } from "@/grid";
+
+export type HintType = "NO_INFORMATION" | "ABSENT" | "CORRECT" | "MISPLACED";
+
+export interface VirtualKeyboardLine {
+  keys: VirtualKeyboardKey[];
+}
+
+export interface VirtualKeyboardKey {
+  value: string;
+  letterHint?: HintType;
+}
+
+const { grid, currentLine, handleLetter, handleEnter, handleBackspace } =
+  useGrid();
+
+const firstLineKeys = ["A", "Z", "E", "R", "T", "Y", "U", "I", "O", "P"];
+const secondLineKeys = ["Q", "S", "D", "F", "G", "H", "J", "K", "L", "M"];
+const thirdLineKeys = ["W", "X", "C", "V", "B", "N"];
+
+const keyboardLines: VirtualKeyboardLine[] = reactive(
+  [firstLineKeys, secondLineKeys, thirdLineKeys].map((lineKeys) => ({
+    keys: lineKeys.map(
+      (letter) =>
+        ({
+          value: letter,
+          letterHint: "NO_INFORMATION",
+        } as VirtualKeyboardKey)
+    ),
+  }))
+);
+
+watch(currentLine, (currentLine) => {
```

Comment:
J'ai un warning sonarlint `'currentLine' is already declared in the upper scope.`

Reply:
J'ai corrigé. Pour plus de clarté j'ai utilisé la notation avec newValue et oldValue, comme ça j'appelle `displayHints(oldValue)` plutôt que d'appeler `displayHints(currentLine - 1)`, ce qui est plus porteur de sens selon moi.
Ca a juste un léger inconvénient, c'est que le paramètre newValue n'est pas utilisé, mais pour moi c'est pas gênant.